src/components/screens/UserProfile/UserPostsScreen.tsx

```diff
@@ -10,21 +11,51 @@ import LoadingErrorView from "../../common/Loading/LoadingErrorView";
 import NotFoundView from "../../common/Loading/NotFoundView";
 import RefreshControl from "../../common/RefreshControl";
 
+interface IRouteParams {
+  fullUsername?: string;
+  isSavedPosts?: boolean;
+}
+
 interface IProps {
-  route: any;
+  route: Route<"UserPostsScreen", IRouteParams>;
 }
 
 function UserPostsScreen({ route }: IProps) {
   const profile = useProfile(false, route?.params?.fullUsername);
 
   const theme = useTheme();
 
+  const isSavedPosts = React.useMemo(
```

Comment:
isSavedPosts doesn't ever change so we shouldn't need to use a memo here.

Reply:
Depends. Assuming the screen component is cached this value will react to any change in it route. This might be the case when you are switching from profile posts to saved profile posts. Not entirely sure though. I did just kinda use it as shorthand that is reactive to changes of route